tests/integration/api_service_test.py

```diff
@@ -196,7 +196,11 @@ def test_create_service_with_resources_constraints(self):
         container_spec = docker.types.ContainerSpec(BUSYBOX, ['true'])
         resources = docker.types.Resources(
             cpu_limit=4000000, mem_limit=3 * 1024 * 1024 * 1024,
-            cpu_reservation=3500000, mem_reservation=2 * 1024 * 1024 * 1024
+            cpu_reservation=3500000, mem_reservation=2 * 1024 * 1024 * 1024,
+            generic_resources=[
+                {'DiscreteResourceSpec': {'Kind': 'gpu', 'Value': 1}},
+                {'NamedResourceSpec': {'Kind': 'gpu', 'Value': 'a'}}
```

Comment:
We should check that we're able to retrieve this data in `res_template` below (data sent by the server)

Reply:
Also it's fine to make a separate test for this.

Reply:
Is [L215](https://github.com/BYU-PCCL/docker-py/blob/0629f7bd6c3d96460e42071d54ed9cd7ef1f7d69/tests/integration/api_service_test.py#L215) not doing that? As a sanity check I printed out `res_template['Resources']['Reservations']` and it contained the generic resources...

Yeah I wasn't sure if we wanted an separate test function...